src/Tools/CoreFx.Tools/GenerateSlnFile.cs

```diff
@@ -0,0 +1,497 @@
+                            currentProject.configurations = ReadConfigurations(configurationProps);
+                            currentProject.Type = projectType.Key;
+                            List<SlnProject> projectGuids = new List<SlnProject>();
+                            if (mapOfProjectsToTypes.ContainsKey(projectType.Key))
+                            {
+                                mapOfProjectsToTypes.TryGetValue(projectType.Key, out projectGuids);
+                                projectGuids.Add(currentProject);
+                            }
+                            else
+                            {
+                                projectGuids.Add(currentProject);
+                                mapOfProjectsToTypes[currentProject.Type] = projectGuids;
+                            }
+                            //ProjectTypeGuids for different projects, pulled from the Visual Studio regkeys
+                            //TODO: Clean up or map these to actual projects, this is fragile
+                            string slnGuid = "FAE04EC0-301F-11D3-BF4B-00C04F79EFBC"; // Windows (C#)
+                            if (projFile.Contains("VisualBasic.vbproj"))
+                            {
+                                slnGuid = "F184B08F-C81C-45F6-A57F-5ABD9991F28F"; //Windows (VB.NET)
+                            }
+                            if (projFile.Contains("TestNativeService")) //Windows (Visual C++) 
+                            {
+                                slnGuid = "8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942";
+                            }
+                            if (projFile.Contains("WebServer.csproj")) //Web Application
+                            {
+                                slnGuid = "349C5851-65DF-11DA-9384-00065B846F21";
+                            }
+
+                            string projectName = Path.GetFileNameWithoutExtension(projFile);
+                            string relativePathFromCurrentDirectory = projFile.Replace(slnFile + Path.DirectorySeparatorChar, "");
+                            string projectDefn =
+                                $"Project(\"{{{slnGuid}}}\") = \"{projectName}\", \"{relativePathFromCurrentDirectory}\", \"{{{currentProject.projectId}}}\"\r\n";
+
+                            string projectDependency = string.Empty;
+                            if (currentProjectType.DependsOn != null)
+                            {
+                                foreach (var dependency in currentProjectType.DependsOn)
+                                {
+                                    if (mapOfProjectTypes[dependency].InUse)
+                                    {
+                                        List<SlnProject> depProjects = null;
+                                        mapOfProjectsToTypes.TryGetValue(dependency, out depProjects);
```

Comment:
It took me a second to realize that this works because of the way we are enumerating the different folders ("project types"). Might be worth mentioning in a comment (or eliminating that timing concern some other way).